React/Views/RCTModalHostViewManager.m

```diff
@@ -92,7 +98,13 @@ - (void)dismissModalHostView:(RCTModalHostView *)modalHostView
   if (_dismissalBlock) {
     _dismissalBlock([modalHostView reactViewController], viewController, animated, nil);
   } else {
-    [viewController.presentingViewController dismissViewControllerAnimated:animated completion:nil];
+    self.dismissWaitingBlock = ^{
+      [viewController.presentingViewController dismissViewControllerAnimated:animated completion:nil];
```

Comment:
Can we try calling `dismissViewControllerAnimated:completion` on viewController instead of `presentingViewController`? I would try it myself but I can't reproduce the issue.

Reply:
I have tried calling `dismissViewControllerAnimated:completion:` on viewController, but it can't fix this issue.
As long as the `presentingViewController` of viewController was nil, dismissing will fail.

Reply:
thanks.